clients/cache/inmemory.go

```diff
@@ -0,0 +1,97 @@
+package cache
+
+import (
+	"context"
+	"sync"
+	"time"
+)
+
+// InMemoryCache is an in-memory implementation of the Cache interface.
+type InMemoryCache struct {
+	data  map[string]cacheItem
+	mutex sync.RWMutex
+}
+
+// Ensure InMemoryCache implements the Cache interface.
+var _ Cache = (*InMemoryCache)(nil)
+
+// cacheItem represents an item stored in the cache.
+type cacheItem struct {
+	data       []byte
+	expiration time.Time
+}
+
+// NewInMemoryCache creates a new instance of InMemoryCache.
+func NewInMemoryCache() *InMemoryCache {
+	return &InMemoryCache{
+		data: make(map[string]cacheItem),
+	}
+}
+
+// Set sets the value of a key in the cache.
+func (c *InMemoryCache) Set(
+	ctx context.Context,
+	key string,
+	data []byte,
+	expiration time.Duration,
+) error {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	expiry := time.Now().Add(expiration)
+
+	if expiration == 0 {
+		// 100 years in the future to prevent expiry
+		expiry = time.Now().AddDate(100, 0, 0)
```

Comment:
make this configurable via an environment variable that the service reads and that is passed in as config when creating the cache

Reply:
you mean instead of hardcoding `100 years` constant? make it configurable?

Reply:
@galxy25 ?

Reply:
yeah, but now that I realize this cache is only used for test code, not an important change